utils/providers.py

```diff
@@ -264,7 +266,7 @@ def setup_provider_by_name(provider_name, *args, **kwargs):
     return setup_provider(get_provider_key(provider_name), *args, **kwargs)
 
 
-def setup_providers(prov_classes=('cloud', 'infra'), validate=True, check_existing=True):
+def setup_providers(prov_classes=['cloud', 'infra'], validate=True, check_existing=True):
```

Comment:
Changed this to a list, when it was called with a single item in a tuple prov_classes was getting set to the item (a string), and was passing the first character of the string as `pclass`